libs/designsystem/src/lib/components/calendar/calendar.component.ts

```diff
@@ -371,18 +405,16 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
     this.calendarHelper.setSelectedDay(newDate.getDate());
   }
 
-  onDateSelected(newDay: CalendarCell) {
+  _onDateSelected(newDay: CalendarCell) {
     if (newDay.isSelectable && newDay.date) {
-      const newDate = this.activeMonth
-        .clone()
-        .date(newDay.date)
-        .toDate();
+      let newDate = new Date(this.activeMonth);
 
-      // emit equivalent date in utc midnight if timezone is not local
-      const dateToEmit =
-        this.timezone === 'local'
-          ? newDate
-          : moment.utc(moment(newDate).format('YYYY-MM-DD')).toDate();
+      if (this.timezone === 'UTC') {
+        newDate = zonedTimeToUtc(this.subtractTimezoneOffset(newDate), this.timeZoneName);
+      }
+
+      newDate.setDate(newDay.date);
```

Comment:
Didn't you change `CalendarCell.date` from number to date? If so, is it still needed to create a new date from `this.activeMonth` + `newDay.date`? 🤔

Reply:
`CalendarCell.date` is still a number, so yes